test/setToken.spec.ts

```diff
@@ -114,42 +112,28 @@ contract("SetToken", (accounts) => {
     });
   };
 
-  const deployStandardSetAndApprove = async (numComponents: number, customUnits: BigNumber[] = []) => {
-    await resetAndDeployComponents(numComponents, customUnits);
-
-    const setTokenTruffle = await SetToken.new(
-      componentAddresses,
-      units,
-      STANDARD_NATURAL_UNIT,
-      TX_DEFAULTS,
-    );
-
-    const setTokenWeb3Contract = web3.eth
-      .contract(setTokenTruffle.abi)
-      .at(setTokenTruffle.address);
-
-    setToken = new SetTokenContract(setTokenWeb3Contract, TX_DEFAULTS);
-
-    const approvePromises = _.map(components, (component) =>
-      component.approve.sendTransactionAsync(setToken.address, UNLIMITED_ALLOWANCE_IN_BASE_UNITS, TX_DEFAULTS),
-    );
-
-    await Promise.all(approvePromises);
-  };
-
   describe("Creation", async () => {
+    let caller: Address = testAccount;
+    
+    async function subject(): Promise<any> {
+      return SetToken.new(
+        componentAddresses,
+        units,
+        naturalUnit,
+        name,
+        symbol,
+        { from: caller },
+      );
+    }
+
     describe(`of Standard Set`, () => {
       beforeEach(async () => {
         await resetAndDeployComponents(2);
       });
 
       it("should work with the correct data", async () => {
```

Comment:
Ideally these are separate test cases, be explicit about what you're testing, or if you're grouping them together.
`it creates a set with the correct symbol`
`it creates a set ...`

Reply:
Mmm. I'm not going to do a full refactor of setToken in this PR. Whoever is working on setToken should be doing so.